spec/lib/nerve/reporter_zookeeper_spec.rb

```diff
@@ -150,67 +150,80 @@
-      end
+      }
+      reporter = Nerve::Reporter::Zookeeper.new(service)
+      str = reporter.send(:encode_child_name, service)
+      JSON.parse(Base64.urlsafe_decode64(str[12...-1])).should == expected
+    end
 
-      it 'encode child name with required fields only' do
-        service = {
-          'instance_id' => 'i-xxxxxx',
-          'use_path_encoding' => true,
-          'host' => '127.0.0.1',
-          'port' => 3000,
-          'zk_hosts' => ['zkhost1', 'zkhost2'],
-          'zk_path' => 'zk_path',
-          'use_path_encoding' => true,
-        }
-        expected = {
-          'name' => 'i-xxxxxx',
-          'host' => '127.0.0.1',
-          'port' => 3000
-        }
-        @reporter = Nerve::Reporter::Zookeeper.new(service)
-        str = @reporter.send(:encode_child_name, service)
-        JSON.parse(Base64.urlsafe_decode64(str[11...-1])).should == expected
-      end
+    it 'encode child name with required fields only' do
+      service = {
+        'instance_id' => 'i-xxxxxx',
+        'use_path_encoding' => true,
+        'host' => '127.0.0.1',
+        'port' => 3000,
+        'zk_hosts' => ['zkhost1', 'zkhost2'],
+        'zk_path' => 'zk_path',
+        'use_path_encoding' => true,
+      }
+      expected = {
+        'name' => 'i-xxxxxx',
+        'host' => '127.0.0.1',
+        'port' => 3000
+      }
+      reporter = Nerve::Reporter::Zookeeper.new(service)
+      str = reporter.send(:encode_child_name, service)
+      JSON.parse(Base64.urlsafe_decode64(str[11...-1])).should == expected
+    end
 
-      it 'encode child name without path encoding' do
-        service = {
-          'instance_id' => 'i-xxxxxx',
-          'host' => '127.0.0.1',
-          'port' => 3000,
-          'zk_hosts' => ['zkhost1', 'zkhost2'],
-          'zk_path' => 'zk_path',
-        }
-        @reporter = Nerve::Reporter::Zookeeper.new(service)
-        expect(@reporter.send(:encode_child_name, service)).to eq('/i-xxxxxx_')
-      end
+    it 'encode child name without path encoding' do
+      service = {
+        'instance_id' => 'i-xxxxxx',
+        'host' => '127.0.0.1',
+        'port' => 3000,
+        'zk_hosts' => ['zkhost1', 'zkhost2'],
+        'zk_path' => 'zk_path',
+      }
+      reporter = Nerve::Reporter::Zookeeper.new(service)
+      expect(reporter.send(:encode_child_name, service)).to eq('/i-xxxxxx_')
+    end
+  end
+
+  context 'create different types of node' do
+    it 'node type should be converted to symbol' do
+      service = subject.merge({'node_type' => 'ephemeral'})
+      reporter = Nerve::Reporter::Zookeeper.new(service)
+      expect(reporter.instance_variable_get(:@mode)).to be_kind_of(Symbol)
+      expect(reporter.instance_variable_get(:@mode)).to eq(:ephemeral)
+    end
 
+    it 'default type of node is :ephemeral_sequential' do
+      reporter = Nerve::Reporter::Zookeeper.new(subject)
+      expect(reporter.instance_variable_get(:@mode)).to eq(:ephemeral_sequential)
```

Comment:
Add a unit test around the type of node that the ZK client receives for the `create` call

Reply:
done